generic_config_updater/patch_sorter.py

```diff
@@ -399,10 +399,10 @@ def _get_paths_recursive(self, config, pattern_tokens, matching_tokens, idx, com
         if token == "*":
             matching_keys = config.keys()
         elif token.startswith("*|"):
-            suffix = token[2:]
+            suffix = token[1:] # the suffix will be `|...`
```

Comment:
Without this change the pattern `Ethernet12|*` would match `Ethernet124|0` and `Ethernet12|0`